### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Q-lds**
PR form dev to main. It includes:

### Dev
- introduced a lowest_cost_bitstrings function
- docplex convertor

### Code 
- Updated setup to fix mathplotlib - networkX install fail
- distinct github actions for dev and main testing
- github actions for publishing
- github actions for code quality and security

### Fixes
- backend seed simulator fix
- cvar bug fix
- fix knapsak generation bug
- gatemap bug fix

### CODE REVIEW

File: openqaoa/workflows/optimizer.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def set_circuit_properties(self, **kwargs):
                 The slope(rate) of linear ramp initialisation of QAOA parameters.
             variational_params_dict: `dict`
                 Dictionary object specifying the initial value of each circuit parameter for the chosen parameterisation, if the `init_type` is selected as `'custom'`.    
+                For example, {'betas': [0.1, 0.2, 0.3], 'gammas': [0.1, 0.2, 0.3]}
```
Comment (Q-lds) [inline openqaoa/workflows/optimizer.py:172]: We should specify this selection is an example of standard-custom

File: docs/source/index.rst [hunk 2]
```diff
@@ -1,49 +1,65 @@
...
 ================
 
 Installing
 ------------
-Clone the git repository:
 
-.. code-block:: bash
-   
-   git clone git@github.com:entropicalabs/openqaoa.git
+You can install the latest version of OpenQAOA directly from PyPi. First, create a virtual environment with *python3.8+* and then simply pip install openqaoa with the following command
```
Comment (Q-lds) [inline docs/source/index.rst:34]: python between 3.8 and 3.10 (included)

File: openqaoa/_version.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "0.0.1dev"
\ No newline at end of file
+__version__ = "0.0.2"
```
Comment (vishal-ph) [inline openqaoa/_version.py:1]: we were on "0.0.1-beta" previously, so shouldn't this be "0.0.1"?

Comment (Q-lds) [reply openqaoa/_version.py:1]: The version will most likely crash with the PyPI test until we reach 0.0.5 :D

Comment (Q-lds) [reply openqaoa/_version.py:1]: But you are right, I will change this PR to 1 and then we see

File: .github/workflows/test_main_linux.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
+name: test_main
+
+# Controls when the workflow will run
+on:
+  push:
```
Comment (Q-lds) [inline .github/workflows/test_main_linux.yml:5]: Remove push

File: openqaoa/backends/qpus/qaoa_qiskit_qpu.py [hunk 5]
```diff
@@ -183,6 +183,7 @@ def get_counts(self, params: QAOAVariationalBaseParams) -> dict:
 
             while api_contact == False:
                 try:
+                    self.job_id = job.job_id()
```
Comment (Q-lds) [inline openqaoa/backends/qpus/qaoa_qiskit_qpu.py:186]: Do we save the jobids for the other backends too?

File: openqaoa/devices.py [hunk 6]
```diff
@@ -256,6 +266,112 @@ def check_connection(self) -> bool:
...
+	available_qpus: `list`
+		When connection to AWS is established, this attribute contains a list
+		of device names which can be used to access the selected device by
+		reinitialising the Access Object with the name of the available device
+		as input to the device_name parameter.
+    """
+    
+    def __init__(self, device_name: str, aws_access_key_id: str, 
+                 aws_secret_access_key: str, aws_region: str, 
+                 s3_bucket_name: str, folder_name: str = 'oq_runs'):
```
Comment (Q-lds) [inline openqaoa/devices.py:287]: Let's call it `openqaoa` if we set a standard. Matching `oq` to `openqaoa` is both trivial and impossible at the same time!

File: openqaoa/workflows/optimizer.py [hunk 7]
```diff
@@ -198,6 +199,8 @@ def set_backend_properties(self, **kwargs):
             n_shots: int
             Optional argument to specify the number of shots required to run QAOA computations
                 on shot-based simulators and QPUs. Defaults to 100.
+            seed_simulator: int
```
Comment (Q-lds) [inline openqaoa/workflows/optimizer.py:202]: Again, I feel this is backend-specific

File: openqaoa/backends/qpus/qaoa_qiskit_qpu.py (hunk 5 above)
Comment (vishal-ph) [reply openqaoa/backends/qpus/qaoa_qiskit_qpu.py:186]: @shahidee44 , maybe you can answer this one?

**Review by vishal-ph — APPROVED** [review]

**Pull request merged**